nova/scheduler/weights/affinity.py

```diff
@@ -55,6 +55,13 @@ def _weigh_object(self, host_state, request_spec):
 class ServerGroupSoftAffinityWeigher(_SoftAffinityWeigherBase):
     policy_name = 'soft-affinity'
     warning_sent = False
+    _SHARD_PREFIX = 'vc-'
+
+    def _is_host_in_aggregate(self, host, aggs):
+        hosts = set(host in agg.hosts for agg in aggs)
```

Comment:
This will create a `{True, False}` set. That's not the intention here, right? You probably want `set(host for agg in aggs for host in agg.hosts)` - which is again 2 for-loops that we wanted to get rid of initially.